crypto/blake2/m_blake2s.c

```diff
@@ -1,61 +1,34 @@
 /*
- * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
  *
  * Licensed under the Apache License 2.0 (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
  * in the file LICENSE in the source distribution or at
  * https://www.openssl.org/source/license.html
  */
 
-/*
- * Derived from the BLAKE2 reference implementation written by Samuel Neves.
- * Copyright 2012, Samuel Neves <sneves@dei.uc.pt>
- * More information about the BLAKE2 hash function and its implementations
- * can be found at https://blake2.net.
- */
-
-#include "internal/cryptlib.h"
-
 #ifndef OPENSSL_NO_BLAKE2
 
-# include <openssl/evp.h>
-# include <openssl/objects.h>
-# include "blake2_locl.h"
+# include <stddef.h>
+# include <openssl/obj_mac.h>
 # include "internal/evp_int.h"
 
-static int init(EVP_MD_CTX *ctx)
-{
-    BLAKE2S_PARAM P;
-    blake2s_param_init(&P);
-    return BLAKE2s_Init(EVP_MD_CTX_md_data(ctx), &P);
-}
-
-static int update(EVP_MD_CTX *ctx, const void *data, size_t count)
-{
-    return BLAKE2s_Update(EVP_MD_CTX_md_data(ctx), data, count);
-}
-
-static int final(EVP_MD_CTX *ctx, unsigned char *md)
-{
-    return BLAKE2s_Final(md, EVP_MD_CTX_md_data(ctx));
-}
-
 static const EVP_MD blake2s_md = {
     NID_blake2s256,
     0,
-    BLAKE2S_DIGEST_LENGTH,
+    32,
     0,
-    init,
-    update,
-    final,
```

Comment:
Because the init/update/final are no lorger used.
Eventually all of the const EVP_MD structures will look like this.. (After the deprecated path can be removed).

Reply:
There are a number of conditions that we check at the start of EVP_DigestInit_ex that could cause us to go down the legacy path, even for digests that would otherwise come from a provider, so I think there could be some scenarios where these functions are still used.

Reply:
probably not for blake though?

Reply:
I think it could occur if an an EVP_MD_CTX that previously had an engine associated with it (so ctx->engine is set) was then reused for a blake operation.